lib/WebService/Async/CustomerIO.pm

```diff
@@ -0,0 +1,320 @@
+package WebService::Async::CustomerIO;
+
+use strict;
+use warnings;
+
+our $VERSION = '0.001';
+
+=head1 NAME
+
+WebService::Async::CustomerIO - unofficial support for the Customer.io service
+
+=head1 SYNOPSIS
+
+=head1 DESCRIPTION
+
+=cut
+
+use parent qw(IO::Async::Notifier);
+
+use mro;
+use Syntax::Keyword::Try;
+use Future;
+use Net::Async::HTTP;
+use Carp qw();
+use JSON::MaybeUTF8 qw(:v1);
+
+use WebService::Async::CustomerIO::Customer;
+use WebService::Async::CustomerIO::RateLimiter;
+use WebService::Async::CustomerIO::Trigger;
+
+use constant {
+    TRACKING_END_POINT                => 'https://track.customer.io/api/v1',
+    API_END_POINT                     => 'https://api.customer.io/v1/api',
+    REQUEST_PER_SECOND_LIMIT_TRACKING => 30,
+    REQUEST_PER_SECOND_LIMIT_API      => 10,
+};
+
+=head2 new
+
+Creates a new api client object
+
+Usage: C<< new(%params) -> obj >>
+
+Parameters:
+
+=over 4
+
+=item * C<site_id>
+
+=item * C<api_key>
+
+=back
+
+=cut
+
+sub configure {
+    my ($self, %args) = @_;
+    for my $k (qw(site_id api_key)) {
+        Carp::croak "Missing required argument: $k" unless exists $args{$k};
+        $self->{$k} = delete $args{$k} if exists $args{$k};
+    }
+
+    $self->{tracking_ratelimiter} ||= do {
+        my $rl =WebService::Async::CustomerIO::RateLimiter->new(
+            limit    => REQUEST_PER_SECOND_LIMIT_TRACKING,
+            interval => 1,
+        );
+        $self->add_child($rl);
```

Comment:
We might not have a loop at this point, better to handle this in `->add_to_loop`